components/device.js

```diff
@@ -0,0 +1,112 @@
+const StdLib = require('@doctormckay/stdlib');
+const Helpers = require('./helpers.js');
+const SteamUser = require('../index.js');
+
+/**
+ * Get authorized devices.
+ * @param {boolean} [inclTokens=false] - If true, include canceled devices
+ * @param {function} [callback]
+ * @returns {Promise}
+ */
+SteamUser.prototype.getOwnAuthorizedDevices = function(includeCanceled, callback) {
+	if (typeof includeCanceled !== 'boolean'){
+		includeCanceled = false;
+	}
+
+	return StdLib.Promises.callbackPromise(null, callback, true, (resolve, reject) => {
+		this._sendUnified('DeviceAuth.GetOwnAuthorizedDevices#1', {
+			'steamid': this.steamID.getSteamID64(),
+			includeCanceled
+		}, (body, hdr) => {
+			let err = Helpers.eresultError(hdr.proto);
+			if (err) {
+				return reject(err);
+			}
+			resolve(body.devices);
+		});
+	});
+};
+
+/**
+ * Authorize local device.
+ * @param {string} deviceName
+ */
+SteamUser.prototype.authorizeLocalDevice = function(deviceName){
+	if(!deviceName){
+		throw new Error('The \'deviceName\' argument is required.');
+	}
+
+	this._send(SteamUser.EMsg.ClientAuthorizeLocalDeviceRequest, {
```

Comment:
No promise/callback here?

Reply:
When "ClientAuthorizeLocalDeviceRequest" is completed, an "authorizedLocalDevice" event is triggered

the same goes for "ClientDeauthorizeDeviceRequest"

both returns the device.

then you can use

```js
client.on('authorizedLocalDevice', function (device) {
  // console.log(device);
}
```